org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WRecordIDDialog.java

```diff
@@ -0,0 +1,196 @@
+		this.editor = editor;
+		this.tableIDGridField = tableIDGridField;
+		if(editor.getValue() instanceof String)
+			this.value = Integer.valueOf((String)editor.getValue());
+		else
+			this.value = (Integer)editor.getValue();
+		
+		init();
+	}
+	
+	private void init() {
+		Vlayout vLayout = new Vlayout();
+		labelsDiv = new Div();
+		fieldsDiv = new Div();
+		contentDiv = new Div();
+		confirmPanelDiv = new Div();
+    	okBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_OK, true, true);;
+    	cancelBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_CANCEL, true, true);;
+    	tableIDEditorCopy = new WTableDirEditor("AD_Table_ID", false, false, true, tableIDGridField.getLookup(), true);
+		tableIDEditorCopy.setValue(tableIDGridField.getValue());
+		MLookup recordsLookup = getRecordsLookup();
+    	recordsEditor = recordsLookup != null ? new WSearchEditor("Record_ID", false, false, true, recordsLookup) : new WSearchEditor(editor.getGridField());
+    	parentTextBox = new Textbox();
+    	
+		setPage(page);
+		setClosable(true);
+		setTitle(Msg.getMsg(Env.getCtx(), "ChooseRelatedRecord"));
+		setPosition("center");
+		setSclass("recordid-dialog");
+		doHighlighted();
+		
+		contentDiv.setSclass("recordid-dialog-content");
+		confirmPanelDiv.setSclass("recordid-dialog-confirm");
+		
+		parentTextBox.setReadonly(true);
+		parentTextBox.setValue(WRecordIDEditor.getIdentifierColumns(
+				editor.getGridField().getGridTab().getAD_Table_ID(), editor.getGridField().getGridTab().getRecord_ID()));
+		
+		tableIDEditorCopy.getComponent().addEventListener(Events.ON_SELECT, this);
+		
+		recordsEditor.setValue(value);
+		
+		okBtn.addEventListener(Events.ON_CLICK, this);
+		okBtn.setSclass("recordid-dialog-confirm");
+		cancelBtn.addEventListener(Events.ON_CLICK, this);
+		cancelBtn.setSclass("recordid-dialog-confirm");
+		
+		Text text = new Text(Msg.getMsg(Env.getCtx(), "Parent"));
+		labelsDiv.appendChild(text);
+		text = new Text(Msg.getMsg(Env.getCtx(), "Table"));
+		labelsDiv.appendChild(text);
+		text = new Text(Msg.getMsg(Env.getCtx(), "Record"));
+		labelsDiv.appendChild(text);
+		labelsDiv.setSclass("recordid-dialog-labels");
+		
+		fieldsDiv.setSclass("recordid-dialog-fields");
+		
+		fieldsDiv.appendChild(parentTextBox);
+		fieldsDiv.appendChild(tableIDEditorCopy.getComponent());
+		fieldsDiv.appendChild(recordsEditor.getComponent());
+		
+		contentDiv.appendChild(labelsDiv);
+		contentDiv.appendChild(fieldsDiv);
+		
+		confirmPanelDiv.appendChild(okBtn);
+		confirmPanelDiv.appendChild(cancelBtn);
+		
+		vLayout.appendChild(contentDiv);
+		vLayout.appendChild(confirmPanelDiv);
+		
+		appendChild(vLayout);
+	}
+	
+	@Override
+	public void onEvent(Event event) throws Exception {
+		if(event.getName().equalsIgnoreCase(Events.ON_CLICK)) {
+			if(event.getTarget().equals(okBtn)) {
+				value = (Integer) recordsEditor.getValue();
+				tableIDGridField.getGridTab().setValue("AD_Table_ID", tableIDEditorCopy.getValue());
+				editor.setValue(value);
```

Comment:
should validate here either both AD_Table_ID and Record_ID is fill with a valid value or both is null.